source/common/common/logger.cc

```diff
@@ -55,29 +58,52 @@ void DelegatingLogSink::set_formatter(std::unique_ptr<spdlog::formatter> formatt
 
 void DelegatingLogSink::log(const spdlog::details::log_msg& msg) {
   absl::ReleasableMutexLock lock(&format_mutex_);
-  if (!formatter_) {
-    lock.Release();
-    sink_->log(absl::string_view(msg.payload.data(), msg.payload.size()));
-    return;
-  }
+  absl::string_view msg_view = absl::string_view(msg.payload.data(), msg.payload.size());
 
+  // This memory buffer must exist in the scope of the entire function,
+  // otherwise the string_view will refer to memory that is already free.
   fmt::memory_buffer formatted;
-  formatter_->format(msg, formatted);
+  if (formatter_) {
+    formatter_->format(msg, formatted);
+    msg_view = absl::string_view(formatted.data(), formatted.size());
+  }
   lock.Release();
-  sink_->log(absl::string_view(formatted.data(), formatted.size()));
+
+  if (should_escape_) {
+    sink_->log(escapeLogLine(msg_view));
+  } else {
+    sink_->log(msg_view);
+  }
+}
+
+std::string DelegatingLogSink::escapeLogLine(absl::string_view msg_view) {
+  // Split the actual message from the trailing whitespace.
+  auto eol_it = std::find_if_not(msg_view.rbegin(), msg_view.rend(), absl::ascii_isspace);
```

Comment:
nit: I'd use absl::string_view explicitly here rather than 'auto', and you don't need to suffix the variable names below with "_view". See pros/cons of https://google.github.io/styleguide/cppguide.html#Type_deduction

Reply:
Done. Thanks for the link to the reading, good to learn :)